tests/window_management_tests/test_minimal_window_manager.cpp

```diff
@@ -0,0 +1,366 @@
+ * it under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <miral/minimal_window_manager.h>
+#include <mir/scene/session.h>
+#include <mir/wayland/weak.h>
+#include <mir/scene/surface_stack.h>
+#include <mir/scene/surface.h>
+#include <mir/executor.h>
+#include <mir/events/event_builders.h>
+#include <mir/events/pointer_event.h>
+#include <mir/compositor/buffer_stream.h>
+#include <mir/graphics/buffer_basic.h>
+#include <mir/test/doubles/stub_buffer_stream.h>
+#include <mir/shell/abstract_shell.h>
+
+#include "mir_test_framework/window_management_test_harness.h"
+#include <linux/input.h>
+
+namespace ms = mir::scene;
+namespace msh = mir::shell;
+namespace geom = mir::geometry;
+namespace mg = mir::graphics;
+namespace mev = mir::events;
+using namespace testing;
+
+MATCHER_P(LockedEq, value, "")
+{
+    return arg.lock() == value;
+}
+
+class CreateappVerifier : public mir_test_framework::WindowManagementVerifier
+{
+public:
+    MOCK_METHOD1(advise_new_app, void(miral::ApplicationInfo&));
+    MOCK_METHOD1(advise_new_window, void(miral::WindowInfo const&));
+    MOCK_METHOD1(advise_delete_window, void(miral::WindowInfo const&));
+};
+
+class MinimalWindowManagerTest : public mir_test_framework::WindowManagementTestHarness
+{
+public:
+    std::shared_ptr<mir_test_framework::WindowManagementVerifier> get_verifier() override
+    {
+        if (!verifier)
+            verifier = std::make_shared<CreateappVerifier>();
+        return verifier;
+    }
+
+    mir_test_framework::WindowManagementPolicyBuilder get_builder() override
+    {
+        return [&](miral::WindowManagerTools const& tools)
+        {
+            return std::make_unique<miral::MinimalWindowManager>(tools);
+        };
+    }
+
+    std::vector<mir::geometry::Rectangle> get_output_rectangles() override
+    {
+        return {
+            mir::geometry::Rectangle{{0, 0}, {800, 600}},
+            mir::geometry::Rectangle{{800, 0}, {800, 600}}
+        };
+    }
+
+    std::shared_ptr<CreateappVerifier> verifier;
+};
+
+TEST_F(MinimalWindowManagerTest, advise_new_app_is_called_when_a_app_is_opened)
+{
+    EXPECT_CALL(*verifier, advise_new_app(_));
+    auto const app = open_application("test");
+}
+
+TEST_F(MinimalWindowManagerTest, advise_new_window_is_called_when_a_window_is_created)
+{
+    EXPECT_CALL(*verifier, advise_new_app(_));
+    auto const app = open_application("test");
+    EXPECT_CALL(*verifier, advise_new_window(_));
+    create_window(app, geom::Width{100}, geom::Height{100});
+}
```

Comment:
This is not actually testing `MinimalWindowManager`, is it? As far as I can tell, this is testing `VerifyableWindowManagementPolicy`? (as in: as far as I can tell, the only change to `MinimalWindowManager` that would cause this test to fail is if `MinimalWindowManager::advice_new_app` threw an exception?)